ios/Classes/VideoCompressPlugin.m

```diff
@@ -35,16 +37,54 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
     } else if ([@"deleteAllCache" isEqualToString:call.method]) {
         [Utility deleteFile:[Utility basePath]];
         result(@YES);
+    } else if([@"getMediaInfo" isEqualToString:call.method]) {
+        NSString *path = args[@"path"];
+        [self getMediaInfo:path result:result];
     } else {
         result(FlutterMethodNotImplemented);
     }
 }
 
 - (NSDictionary *)getMediaInfoJson:(NSString *)path {
-    NSDictionary* dictionary = @{ @"path": [Utility excludeFileProtocol:path] };
+    NSURL *url = [Utility getPathUrl:path];
+    AVURLAsset *asset = [AvController getVideoAsset:url];
+    AVAssetTrack *track = [AvController getTrack:asset];
+
+    AVPlayerItem *playerItem = [AVPlayerItem playerItemWithURL:url];
+    AVAsset *metadataAsset = playerItem.asset;
+
+    NSInteger orientation = [AvController getVideoOrientation:path];
+
+    NSString *title = [AvController getMetaDataByTag:metadataAsset key:@"title"];
+    NSString *author = [AvController getMetaDataByTag:metadataAsset key:@"author"];
+
+    double duration = asset.duration.value;
+    double filesize = track.totalSampleDataLength;
+
+    CGSize size = CGSizeApplyAffineTransform(track.naturalSize, track.preferredTransform);
+
+    double width = fabs(size.width);
+    double height = fabs(size.height);
+
+    NSDictionary* dictionary = @{
+        @"path": [Utility excludeFileProtocol:path],
+        @"title": title,
+        @"author": author,
+        @"width": [NSNumber numberWithDouble:width],
+        @"height": [NSNumber numberWithDouble:height],
+        @"duration": [NSNumber numberWithDouble:duration],
+        @"filesize": [NSNumber numberWithDouble:filesize],
+        @"orientation": [NSNumber numberWithInteger:orientation],
+    };
     return dictionary;
 }
 
+- (void)getMediaInfo:(NSString *)path result:(FlutterResult)result {
+    NSDictionary *json = [self getMediaInfoJson: path];
+    NSString *string = [Utility keyValueToJson: json];
```

Comment:
no need for space before parameter in these 2 selector calls (e.g. [self getMediaInfoJson:path] is fine)